### PR TITLE
wip nextjs@15 with storybook-webpack

### PR DESCRIPTION
**Pull request by bmuenzenmeyer**
## Description

during collab summit, @ovflowd and I experimented with unblocking #7155 by no longer using storybook next

## Validation

- sidebar is broken, react.cache is a problem yet
- we want to see chromatic differences

### Check List

- [x] I have read the [Contributing Guidelines](https://github.com/nodejs/nodejs.org/blob/main/CONTRIBUTING.md) and made commit messages that follow the guideline.
- [x] I have run `npm run format` to ensure the code follows the style guide.
- [x] I have run `npm run test` to check if all tests are passing.
- [x] I have run `npx turbo build` to check if the website builds without errors.
- [x] I've covered new added functionality with unit tests if necessary.

### CODE REVIEW

**Comment by ovflowd** [comment]
@bmuenzenmeyer didn't we fix sidebar with the client-context mock? 👀

File: apps/site/.storybook/__mocks__/client-context.mjs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+// todo sidebar is broken - react cache import is a problem
+
+export const getClientContext = () => {
+  return {
```
Comment (ovflowd) [inline apps/site/.storybook/__mocks__/client-context.mjs]: Nit: use ({ }) to remove the need of a return statement

File: apps/site/.storybook/main.ts [hunk 2]
```diff
@@ -1,31 +1,60 @@
...
+    jsc: {
+      transform: {
+        react: {
+          runtime: 'automatic',
+        },
+      },
+    },
+  }),
   previewBody:
+    '<link href="https://fonts.googleapis.com/css2?family=IBM+Plex+Mono&family=Open+Sans:ital,wght@0,300..800;1,300..800&display=swap" rel="stylesheet">' +
```
Comment (ovflowd) [inline apps/site/.storybook/main.ts:16]: Maybe worth adding inline line docs explaining what these additions are (including the style ones below) and why they are needed

**Comment by bmuenzenmeyer** [comment]
There was another error beyond what we resolved with react.cache.
I can play with it at the airport.  Need sleep now

**Comment by ovflowd** [comment]
@bmuenzenmeyer I did the work for you, have a good night <3

**Pull request merged**